src/general/MainMenu.tscn

```diff
@@ -10,6 +10,7 @@
 [ext_resource path="res://assets/textures/gui/BG_Menu01.png" type="Texture" id=8]
 [ext_resource path="res://assets/textures/gui/BG_Menu03.png" type="Texture" id=9]
 [ext_resource path="res://assets/textures/gui/logo.png" type="Texture" id=10]
+[ext_resource path="res://src/general/DialogInputHelper.cs" type="Script" id=11]
```

Comment:
Instead of adding this Helper on various scenes, add this helper as an AutoLoad (Godot: Project>Project Settings...>AutoLoad).

This makes sure that there is only one instance of your class, which also allows you to remove all those isDuplicate checks

Reply:
Then the _Ready() only gets called once, making it miss all other dialogs.